chainer/functions/pooling/unpooling_2d.py

```diff
@@ -0,0 +1,57 @@
+import numpy
+
+from chainer import cuda
+from chainer.functions.pooling import pooling_2d
+from chainer.utils import conv
+
+
+class Unpooling2D(pooling_2d.Pooling2D):
+
+    """Unpooling over a set of 2d planes."""
+
+    def __init__(self, ksize, stride=None, pad=0,
+                 outsize=None, cover_all=True, use_cudnn=True):
+        super(Unpooling2D, self).__init__(
+            ksize, stride, pad, cover_all, use_cudnn)
+        self.outh, self.outw = (None, None) if outsize is None else outsize
+
+    def forward_cpu(self, x):
+        h, w = x[0].shape[2:]
+        if self.outh is None:
+            self.outh = conv.get_deconv_outsize(h, self.kh, self.sy, self.ph)
+        if self.outw is None:
+            self.outw = conv.get_deconv_outsize(w, self.kw, self.sx, self.pw)
+        col = numpy.tile(x[0][:, :, numpy.newaxis, numpy.newaxis],
```

Comment:
We can unify cpu and gpu code by `x[0][:, :, None, None]`. `cupy.newaxis` eqauls to `numpy.newaxis`, which is equals to `None`.